src/main/java/seedu/address/model/student/UniqueStudentList.java

```diff
@@ -241,52 +488,174 @@ create a class from the start time and check to see if it exceeds the startTime
     }
 
-    private static Class findAvailableClassWithSingleRecord(TimeRange tr, LocalDate currDate, List<Student> list) {
-        Class newClass;
-        Class classToCompare = list.get(0).getAClass();
+    /**
+     * Find next available class when there is only one class.
+     * @param tr the timeRange.
+     * @param currDate the current date.
+     * @param list the list of Classes.
+     * @param currTime the currTime.
+     * @return the next available class.
+     */
+    public static Class findAvailableClassWithSingleRecord(TimeRange tr, LocalDate currDate, List<Class> list,
+                                                            LocalTime currTime) {
+        Class newClass = null;
+        Class classToCompare = list.get(0);
         // When the startTimeRange is before the earliest slot
         assert classToCompare.endTime != null;
         assert classToCompare.startTime != null;
 
-        if (classToCompare.endTime.compareTo(tr.startTimeRange) <= 0
-                || (classToCompare.startTime.compareTo(tr.startTimeRange) >= 0
-                && tr.startTimeRange.plusMinutes(tr.duration).compareTo(classToCompare.startTime) <= 0)) {
-            newClass = new Class(currDate, tr.startTimeRange,
-                    tr.startTimeRange.plusMinutes(tr.duration));
-        } else if (classToCompare.startTime.compareTo(tr.endTimeRange) >= 0
-                || (classToCompare.endTime.plusMinutes(tr.duration).compareTo(tr.endTimeRange) <= 0)) {
-            // When the startTimeRange is after the earliest slot
-            newClass = new Class(currDate, classToCompare.endTime,
-                    classToCompare.endTime.plusMinutes(tr.duration));
+        LocalTime startTimeFromTr = tr.startTimeRange;
+        LocalTime endTimeFromTr = startTimeFromTr.plusMinutes(tr.duration);
+        LocalTime startTimeFromClass = classToCompare.endTime;
+        LocalTime endTimeFromClass = startTimeFromClass.plusMinutes(tr.duration);
+        LocalTime startTimeFromCurrTime = currTime;
+        LocalTime endTimeFromCurrTime = startTimeFromCurrTime.plusMinutes(tr.duration);
+
+        if (!classToCompare.date.equals(currDate)) {
+            newClass = new Class(currDate, startTimeFromTr, endTimeFromTr);
+        } else if (currTime.compareTo(tr.endTimeRange) >= 0) {
+            // if the currentTime is after and including the end of the time range, just go to the next day
+            newClass = new Class(currDate.plusDays(1), startTimeFromTr, endTimeFromTr);
+        } else if (currTime.compareTo(tr.startTimeRange) <= 0) {
+            // if the currentTime is before and including the start time range
+            if (classToCompare.endTime.compareTo(tr.startTimeRange) <= 0) {
+                // the sitaution where the endTime of the class is before the tr.startTimeRange
+                newClass = new Class(currDate, startTimeFromTr, endTimeFromTr);
+            } else if (classToCompare.endTime.compareTo(tr.startTimeRange) > 0
```

Comment:
At the current timing of "1730", with only one class set to be "2022-11-06 1630-1830", when calling the "avail 1630-1930 61", it will give a result of "Available date is: 2022-11-06 1830-1931" which is out of the given range.

Reply:
Fantastic catch! Realised out I missed out a check in my code. Thanks for pointing it out!